File: C#/14.cs
public class Solution {
    public string LongestCommonPrefix(string[] strs) {
        if(strs == null || strs.Length == 0) {
            return "";
        }
        
        string prefix = strs[0];
        
        for(int i = 1; i < strs.Length; i++) {
            while(strs[i].IndexOf(prefix) != 0) {
                prefix = prefix.Substring(0, prefix.Length - 1);
                if(string.IsNullOrEmpty(prefix)) {
                    return "";
                }
            }
        }
        
        return prefix;
    }
}

File: python/14.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""
        prefix = strs[0]
        for word in strs[1:]:
            while word.find(prefix) != 0:
                prefix = prefix[:-1]
                if not prefix:
                    return ""
        return prefix